src/primitives/cylindersurface.jl

```diff
@@ -61,7 +64,7 @@ end
 
 axis(c::CylinderSurface) = Line(c.bot(0, 0), c.top(0, 0))
 
-function isright(c::CylinderSurface{T}) where {T}
+function isright(c::CylinderSurface{P,T}) where {P,T}
```

Comment:
Is T playing the role of L in this method? Let's use the same type parameter name for consistency.